Yank/storage.py

```diff
@@ -0,0 +1,451 @@
+#!/usr/bin/env/python
+
+# =============================================================================
+# MODULE DOCSTRING
+# =============================================================================
+
+"""
+Classes that store arbitrary NetCDF (or other) options and describe how to handle them
+
+"""
+
+# =============================================================================
+# GLOBAL IMPORTS
+# =============================================================================
+
+import abc
+import copy
+import netCDF4 as nc
+
+import numpy as np
+from simtk import openmm, unit
+
+from . import utils
+
+ABC = abc.ABCMeta('ABC', (object,), {})  # compatible with Python 2 *and* 3
+
+# =============================================================================
+# MODULE FUNCTIONS
+# =============================================================================
+
+# =============================================================================
+# CUSTOM EXCEPTIONS
+# =============================================================================
+
+# =============================================================================
+# GENERIC SAVABLE DATA
+# =============================================================================
+
+
+class StorageInterfaceDirVar(object):
```

Comment:
I don't know how the internals will be, but is there a reason why we can't have two separate `StorageInterfaceDir` and `StorageInterfaceVar(StorageInterfaceDir)` classes? This way you could avoid dynamically locking methods.